eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -133,66 +131,56 @@ stages:
-
-                          # Run vcpkg x-add-version to add version to vcpkg file
-                          Write-Host "./bootstrap-vcpkg.bat"
-                          ./bootstrap-vcpkg.bat
-                          Write-Host "./vcpkg.exe x-add-version ${{ artifact.VcpkgPortName }}"
-                          ./vcpkg.exe x-add-version ${{ artifact.VcpkgPortName }}
-
-                          # Amend commit to include results of x-add-version
-                          Write-Host "git status"
+                      # Check out the PR branch if it's already in remote. 
+                      # Ignore failures.
+                      - pwsh: |
+                          $ErrorActionPreference = "Continue"
+                          git checkout "origin/$(PrBranchName)" 2>&1 | Out-Null
+                          $LASTEXITCODE = 0 # This ignores any error from git checkout
                           git status
-                          Write-Host "git add -A"
-                          git add -A
-                          Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" commit --amend -m `"[${{ artifact.VcpkgPortName }}] Update to $(PackageVersion)`""
-                          git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" commit --amend -m "[${{ artifact.VcpkgPortName }}] Update to $(PackageVersion)"
-
-                          # Work around create-pull-request.yml logic that 
-                          # checks for changes
-                          Write-Host "##vso[task.setvariable variable=HasChanges]$true"
-                        displayName: Commit changes for PR
+                        displayName: Checkout Previous PRBranch if it exist.
+                        workingDirectory: $(VcpkgWorkspace)
 
+                      - task: Powershell@2
+                        inputs:
+                          pwsh: true
+                          targetType: filePath
+                          filePath: eng/scripts/Update-VcpkgPort.ps1
+                          arguments: >-
+                            -ReleaseArtifactSourceDirectory "$(Pipeline.Workspace)/packages/${{ artifact.Name }}"
+                            -PortDestinationDirectory 'ports/${{ artifact.VcpkgPortName }}'
+                            -VcpkgPortName '${{ artifact.VcpkgPortName }}'
+                            -GitCommitParameters '-c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com"'
+                          workingDirectory: $(VcpkgWorkspace)
+                        displayName: Update vcpkg port
+
+                      # Set $(HasChanges) to $true so that 
+                      # create-pull-request.yml completes the push and PR 
+                      # submission steps
+                      - pwsh: Write-Host "##vso[task.setvariable variable=HasChanges]$true"
```

Comment:
I was going to ask why you are doing this but I think I figured out why which is because you want the commit message to be customized and more then the string we can pass along. I'm good with this approach for now but I'd like us to consider use adding the option to pass a file through instead.

Reply:
I've opened this issue to add this functionality.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/1473